nucliadb_vectors/src/data_point_provider/merge_worker.rs

```diff
@@ -64,16 +65,22 @@ impl Worker {
             tracing::info!("Could not request merge: {}", e);
         }
     }
-    fn work(&self) -> VectorR<()> {
-        while self.work_flag.try_to_start_working().is_err() {}
```

Comment:
hmmm, so it was just running forever here? This might have caused constant cpu spikes checking for lock then?

Reply:
`work_flag` is a mutex defined on the standard library. It has move semantics, so when it goes out of scope or the variable is not live anymore is dropped (therefore releasing the lock). You can look at the documentation [here](https://doc.rust-lang.org/src/std/sync/mutex.rs.html#178)

Reply:
Is not running forever because is trying to acquire the mutex not blocking. This will stall in the case the garbage collection is running at that time, anyhow this line was completely useless because the lock is being dropped as soon as the while ends. You can see an example of this in the [following's code MIR](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021).

Reply:
Okay, I didn't mean forever that way--just until the lock is available it was previously running over and over until it was available. `while self.work_flag.try_to_start_working().is_err() {}`

In any case, you fixed it :)

Reply:
re locks, that makes more sense. And explicitly dropping them out of memory in the code block is an optimization because they would be unlocked automatically anyways when function finished executing then

Reply:
ouh I see, yeah that was too consuming even though it is a separate thread. I though your concern was that two lines after that I try the same thing in the proper way and that may be causing a deadlock (Lucky for me Rust  came to the rescue in this case..).